src/webpack/index.ts

```diff
@@ -75,3 +84,55 @@ function removeMiniCssExtract(config: any) {
     }
   });
 }
+
+function getLibraryName(options: any) {
+  if (options?.customWebpackConfig?.libraryName) {
+    return options.customWebpackConfig.libraryName;
+  }
+
+  const projectName = getProjectNameFromAngularJson();
+  if (projectName) return projectName;
+
+  console.warn(
+    'Warning: single-spa-angular could not determine a library name to use and has used a default value.',
+  );
+  console.info('This may cause issues if this app uses code-splitting or lazy loading.');
+  if (!options) {
+    console.info('You may also need to update extra-webpack.config.json.');
+  }
+  console.info('See <SomeUrl> for information on how to resolve this.');
+
+  return 'angular_single_spa_project';
+}
+
+function getProjectNameFromAngularJson(): string | null {
+  const angularJsonPath = findUp(['angular.json', '.angular.json']);
+  if (!angularJsonPath) return null;
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const angularJson = require(angularJsonPath);
+  if (!angularJson.projects) return null;
+
+  var projects = Object.getOwnPropertyNames(angularJson.projects);
```

Comment:
Wouldn't `Object.keys()` work fine here? JSON files don't have non-enumerable properties, nor do they have prototype inheritance, so I don't think we'd lose anything by using the more common and familiar Object.keys function.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys

Reply:
Yeah `Object.keys` should be fine. I normally reach for `getOwnPropertyNames` out of habit since it is usually the safest option unless I need prototypical inheritance.